spec/models/student_spec.rb

```diff
@@ -0,0 +1,326 @@
+  #     expect(Group.any_instance.stubs(:repo)).to eq(mock_repo)
+  #
+  #     Student.unhide_students(@student_id_list)
+  #   end
+  # end
+  #
+  # context 'A hidden Student' do
+  #     @student = Student.new(:hidden)
+  #
+  #     it 'should not become a member of a grouping' do
+  #       grouping = Grouping.new
+  #       @student.invite(grouping.id)
+  #
+  #       pending_memberships = @student.pending_memberships_for(grouping.assignment_id)
+  #
+  #       expect(pending_memberships).not_to be_nil
+  #       expect(pending_memberships.length).to eq(0)
+  #     end
+  #   end
+
+  context 'A Student' do
+    @student = Student.new
+
+    context 'with an assignment' do
+      @assignment = Assignment.new(group_name_autogenerated: false)
+
+      it 'should not return nil on call to memberships_for' do
+          expect(@student.memberships_for(@assignment.id)).not_to be_nil
+      end
+
+      it 'should return empty when there are no pending memberships' do
+        pending_memberships = @student.pending_memberships_for(@assignment.id)
+
+        expect(pending_memberships).not_to be_nil
+        expect(pending_memberships.length).to eq(0)
+      end
+
+      it 'should return nil when there are no pending groupings' do
+        expect(Student.any_instance.stubs(:pending_groupings_for)).to be_nil
+
+        pending_memberships = @student.pending_memberships_for(@assignment.id)
+        expect(pending_memberships).to be_nil
+      end
+
+      it 'should correctly return if it has accepted groupings' do
+        expect(@student.has_accepted_grouping_for?(@assignment.id)).to be_falsey
+        membership = StudentMembership.new(user: @student, grouping: Grouping.new(assignment: @assignment),
+                                              membership_status: StudentMembership::STATUSES[:inviter])
+        expect(@student.has_accepted_grouping_for?(@assignment.id)).to be true
+      end
+
+      it 'should correctly return the accepted grouping' do
+        expect(@student.accepted_grouping_for(@assignment.id)).to be_nil
+        grouping = Grouping.new(assignment: @assignment)
+        membership = StudentMembership.new({user: @student, grouping: grouping,
+                                             membership_status: StudentMembership::STATUSES[:inviter]})
+        expect(grouping).to eq(@student.accepted_grouping_for(@assignment.id))
+      end
+
+      it 'should raise when creating an autogenerated name group' do
+        expect(@student.create_autogenerated_name_group(@assignment.id)).to raise_error(RuntimeError)
+      end
+    end
+
+    context 'and a grouping' do
+      it 'should be invited to a grouping' do
+        grouping = Grouping.new
+        @student.invite(grouping.id)
+
+        pending_memberships = @student.pending_memberships_for(grouping.assignment_id)
+
+        expect(pending_memberships).not_to be_nil
+        expect(pending_memberships.length).to eq(1)
```

Comment:
Please use a single `expect` statement per test (one `expect` for every `it` context). It makes it easier to determine exactly why a test has passed or failed.